tauri-app/src/lib/ModalVaultWithdraw.svelte

```diff
@@ -0,0 +1,87 @@
+<script lang="ts">
+  import { Button, Modal, Label, Helper } from 'flowbite-svelte';
+  import type { Vault } from '$lib/typeshare/vault';
+  import InputTokenAmount from './InputTokenAmount.svelte';
+
+  export let open = false;
+  export let vault: Vault;
+  let amount: string = '';
+  let amountRaw: bigint = 0n;
+  let amountGTBalance: boolean;
+
+  $: {
+    if (vault.token_vaults) {
+      console.log('amount raw is ', amountRaw);
+      try {
+        amountGTBalance = amountRaw > vault.token_vaults[0].balance;
+      } catch (e) {}
+    }
+  }
+
+  function reset() {
+    amount = '';
+    amountRaw = 0n;
+  }
+</script>
+
+<Modal title="Withdraw from Vault" bind:open outsideclose size="sm" on:close={reset}>
+  <div>
+    <h5 class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white">
+      Vault ID
+    </h5>
+    <p class="break-all font-normal leading-tight text-gray-700 dark:text-gray-400">
+      {vault.id}
+    </p>
+  </div>
+
+  <div>
+    <h5 class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white">
+      Token
+    </h5>
+    <p class="break-all font-normal leading-tight text-gray-700 dark:text-gray-400">
+      {vault.token_vaults && vault.token_vaults[0].token.name}
+    </p>
+  </div>
+
+  <div>
+    <h5 class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white">
+      Balance
+    </h5>
+    <p class="break-all font-normal leading-tight text-gray-700 dark:text-gray-400">
+      {vault.token_vaults && vault.token_vaults[0].balance_display}
+    </p>
+  </div>
+
+  <div class="mb-6">
+    <Label
+      for="amount"
+      class="mb-2 w-full text-xl font-bold tracking-tight text-gray-900 dark:text-white"
+    >
+      Amount
+    </Label>
+    <InputTokenAmount
+      bind:value={amount}
+      bind:valueRaw={amountRaw}
+      symbol={vault.token_vaults ? vault.token_vaults[0].token.symbol : ''}
+      decimals={vault.token_vaults ? vault.token_vaults[0].token.decimals : 16}
```

Comment:
16?

Reply:
actually i think we should fallback to 0 with a clear error/warning that shows to the user that they have to put the raw absolute value in because we _can't_ infer the decimals for them

Reply:
it's resolved in PR #111 -- removed all the conditionals of vault.token_vaults